eel/src/lib.rs

```diff
@@ -450,38 +438,63 @@ impl LightningNode {
 
         let fiat_values = self.get_fiat_values(amount_msat);
 
+        if let Ok(payment) = self
+            .payment_store
+            .lock()
+            .unwrap()
+            .get_payment(invoice_struct.payment_hash())
+        {
+            if payment.payment_type == PaymentType::Receiving {
+                return Err(runtime_error(
+                    RuntimeErrorCode::AlreadyUsedInvoice,
+                    "This invoice was issued by the local node. Please use a different one.",
+                ));
+            }
+
+            if payment.payment_state != PaymentState::Failed {
```

Comment:
I refactored that part of the code a bit. I used a `match` on the `payment_type` but still kept an `if` statement for the `payment_state` because all possibilities except `PaymentState::Failed` mean an error should be returned.

Reply:
I like it :+1: